src/main/java/com/user_service/services/UserAuthenticationService.java

```diff
@@ -71,14 +79,14 @@ public void signInUser(LoginUserDto loginUserDto) {
                     .authFlow(AuthFlowType.ADMIN_USER_PASSWORD_AUTH)
                     .build();
 
-            AdminInitiateAuthResponse response = cognitoIdentityProviderClient.adminInitiateAuth(authRequest);
-            System.out.println("Result Challenge is : " + response.toString() );
-
-        } catch(CognitoIdentityProviderException e) {
-            System.err.println(e.awsErrorDetails().errorMessage());
-            System.exit(1);
+            log.info("User {} signed in", signInUserDto.getEmail());
+            return new SignInResponseDto(cognitoIdentityProviderClient.adminInitiateAuth(authRequest), null);
+        } catch (CognitoIdentityProviderException e) {
+            log.error("Failed to sign in user, error: {}", e.awsErrorDetails().errorMessage());
+            return new SignInResponseDto(null, e.awsErrorDetails().errorMessage());
+        } finally {
```

Comment:
You should be able to use [try-with-resources](https://www.baeldung.com/java-try-with-resources) and avoid manual closing of a client.